vinyldns/resource_record_set.go

```diff
@@ -104,8 +105,48 @@ func resourceVinylDNSRecordSetRead(d *schema.ResourceData, meta interface{}) err
 	if err != nil {
 		return err
 	}
+	recordType := strings.ToLower(rs.Type)
+
+	if recordType == "soa" {
+		return errors.New(recordType + " records are not currently supported by vinyldns")
+	}
 
 	d.Set("name", rs.Name)
+	d.Set("zone_id", rs.ZoneID)
+	d.Set("ttl", rs.TTL)
+	d.Set("type", rs.Type)
+	d.Set("owner_group_id", rs.OwnerGroupID)
+
+	if recordType == "cname" {
+		d.Set("record_cname", rs.Records[0].CName)
+
+		return nil
+	}
+
+	if recordType == "txt" {
+		d.Set("record_text", rs.Records[0].Text)
```

Comment:
@mdb I believe we could get into some trouble here and lose state. As far as I know, multivalue TXT records are legitimate, but they cannot overlap values. This should likely be handled as a set of TXT records.

Reply:
@rhenning Wow - good catch. While this is not a regression specific to this PR, now is arguably a good time to fix it, as I'm already changing the schema and versioning it at 1. I pushed a fix; please let me know what you think about my implementation and attribute renaming to `record_texts` -- is that an awkward name?

Reply:
@mdb While it might be slightly awkward, I'd say that your proposed name of `record_texts` conforms pretty closely to [Vinyl's record model](https://www.vinyldns.io/api/recordset-model). 👍